tests/unit/test_parsers.py

```diff
@@ -544,7 +544,9 @@ def test_parsers(parser_class, raw_file, results_file):
     with open(raw_file, "rb") as file_obj:
         raw_data = file_obj.read()
 
-    parsed_notifications = parser_class().parse(raw_data)
+    parsed_notifications = parser_class().parse(
+        raw_data, parser_class._data_types[0]  # pylint: disable=protected-access
+    )
```

Comment:
Can you help me understand why this change? It feels weird to be passing a private attribute of the `parser_class` into a method on said class - why doesn't the parser just look this up automatically?

Reply:
before this PR, the Parsers were only able to manage one data type: HTML, CSV, text, etc. The LLM Parser supports two types: HTML and text, but it requires a specific transform of the data into text (done [here](https://github.com/networktocode/circuit-maintenance-parser/pull/245/files#diff-b5df3420c13a1de156bbb25c6f41d0448704d43bd9feee7f0bcfe131804f0f8eR281)). 

However, I should change this private access to the public method to get them.